diffusion/callbacks/log_diffusion_images.py

```diff
@@ -38,62 +40,67 @@ class LogDiffusionImages(Callback):
 
     def __init__(self,
                  prompts: List[str],
-                 size: Optional[int] = 256,
+                 size: Optional[Union[Tuple[int, int], int]] = 256,
+                 batch_size: Optional[int] = None,
                  num_inference_steps=50,
                  guidance_scale: Optional[float] = 0.0,
                  rescaled_guidance: Optional[float] = None,
-                 text_key: Optional[str] = 'captions',
                  tokenized_prompts: Optional[torch.LongTensor] = None,
                  seed: Optional[int] = 1138,
                  use_table: bool = False):
         self.prompts = prompts
-        self.size = size
+        self.size = size if isinstance(size, tuple) else (size, size)
+        self.batch_size = len(prompts) if batch_size is None else batch_size
         self.num_inference_steps = num_inference_steps
         self.guidance_scale = guidance_scale
         self.rescaled_guidance = rescaled_guidance
-        self.text_key = text_key
         self.seed = seed
         self.tokenized_prompts = tokenized_prompts
         self.use_table = use_table
 
-    def eval_batch_end(self, state: State, logger: Logger):
-        # Only log once per eval epoch
-        if state.eval_timestamp.get(TimeUnit.BATCH).value == 1:
-            # Get the model object if it has been wrapped by DDP
-            # We need this to access the text keys, tokenizer, and image generation function.
-            if isinstance(state.model, DistributedDataParallel):
-                model = state.model.module
+    def eval_start(self, state: State, logger: Logger):
+        # Get the model object if it has been wrapped by DDP
+        # We need this to access the text keys, tokenizer, and image generation function.
+        if isinstance(state.model, DistributedDataParallel):
+            model = state.model.module
+        else:
+            model = state.model
+
+        if self.tokenized_prompts is None:
```

Comment:
I have always found this a bit ugly. What do you think about not tokenizing the prompts here, and instead letting `model.generate` handle that by passing in string prompts? It would allow us to avoid the check for `model.sdxl` here, and avoid making the assumption that we can call `model.tokenizer` this way. Would require slightly different batching below though.

Reply:
Removed! It's a lot cleaner